File: lecture2_Practise_problem_intermidiate2_right_shift_and_rotate_by_k_placesd.java
public class lecture2_Practise_problem_intermidiate2_right_shift_and_rotate_by_k_placesd {
    public static void main(String[] args) {
        int[] source = {10, 20, 30, 40, 50};
        int k = 3;
        rightShift(source, k);
        print_array(source);
    }
    public static void rightShift(int[] s,int shift) {
        for(int i = s.length-1; i >=shift;i--){
            s[i] = s[i - shift];
        }
        for(int j = 0; j<shift;j++){
            s[j] = 0;
        }
    }
    public static void print_array(int[] y) {
        System.out.println("Address of array Y: "+ y);
        for(int value: y){
            System.out.println(value);
        }
    }
}

File: lecture2_Practise_problem_intermidiate2_right_shift_and_rotate_by_k_placesd.py
#right shifting an array by k places

def rightShift(source,k):

    i=len(source)-1 #pointer at the last index of source

    while(i>=k):

        source[i]=source[i-k]  #shifting elements k places to the right

        i=i-1

    i=0

    while(i<k):

        source[i]=0 #setting first k values to 0

        i=i+1



a=[10,20,30,40,50]

rightShift(a,3)

print(a)